### PR TITLE
Update source for ctapipe to 0.17

### PR DESCRIPTION
**Pull request by maxnoe**
Tests run with the exception for the tests needing calibration files. These require new calibration files produced with ctapipe 0.17.

I intend to solve this hen/egg problem by also opening the PR in lstchain and use the two branches together to produce new calibration files with what already works.

### CODE REVIEW

**Comment by maxnoe** [comment]
@FrancaCassol @rlopezcoto @moralejo I think this is ready to be reviewed, tests are passing

**Review by moralejo — APPROVED** [review]

File: src/ctapipe_io_lst/__init__.py [hunk 1]
```diff
@@ -322,42 +368,47 @@ def datalevels(self):
         return (DataLevel.R0, )
 
     @staticmethod
-    def create_subarray(geometry_version, tel_id=1):
+    def create_subarray(tel_id=1):
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply src/ctapipe_io_lst/__init__.py:371]: I now bundle only a single LSTCam file in the repo with the current (and updated format). So there is no point in the version anymore.

File: src/ctapipe_io_lst/__init__.py [hunk 2]
```diff
@@ -322,42 +368,47 @@ def datalevels(self):
...
         )
 
         tel_descriptions = {tel_id: lst_tel_descr}
 
-        # LSTs telescope position taken from MC from the moment
-        tel_positions = {tel_id: [50., 50., 16] * u.m}
+        # put LST at 0, so that it is at the given position
+        # TODO: is that the right choice for LST + MAGIC Analysis?
+        # or should we use the same relative position and reference point as the MC?
+        tel_positions = {tel_id: [0, 0, 0] * u.m}
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply src/ctapipe_io_lst/__init__.py:405]: Yes, sorry, forgot about this. We should make there reference position configurable and compute these offsets when needed. 

I'll also open an issue with ctapipe to support giving the telescope positions in the subarray as `EarthLocation`.

**Review by FrancaCassol — APPROVED** [review]

File: src/ctapipe_io_lst/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (maxnoe) [reply src/ctapipe_io_lst/__init__.py:405]: @jsitarek Maybe you can provide comments her, is this important to get right for the joint analysis?

[the remark this answers is not in the document]
Comment (maxnoe) [reply src/ctapipe_io_lst/__init__.py]: Ok, the positions are now calculated to the reference position calculated from official lon/lat values using abelardo's method (area weighted average with mirror areas of 23² m² for LST and 17² m² for MAGIC-{1,2}).

The calculated positions are within 0.6 m of the monte carlo positions.

**Comment by maxnoe** [comment]
Ok, the positions are now calculated to the reference position calculated from official lon/lat values using abelardo's method (area weighted average with mirror areas of 23² m² for LST and 17² m² for MAGIC-{1,2}).

The calculated positions are within 0.6 m of the monte carlo positions.

**Comment by maxnoe** [comment]
For future reference:

```python

from astropy.coordinates import EarthLocation, AltAz, ITRS
from astropy.time import Time
import astropy.units as u
import matplotlib.pyplot as plt
from pyproj import CRS, Transformer
import numpy as np
from ctapipe.io import EventSource


locations = {
    "LST-1": dict(
        loc=EarthLocation(lon=-17.89149701 * u.deg, lat=28.76152611 * u.deg, height=2185 * u.m),
        rough_area=23**2 * u.m**2,
        tel_id=1,
    ),
    "MAGIC-1": dict(
        loc=EarthLocation(lon=-17.8900665372 * u.deg, lat=28.7619439944 * u.deg, height=2184.763 * u.m),
        rough_area=17**2 * u.m**2,
        tel_id=2,
    ),
    "MAGIC-2": dict(
        loc=EarthLocation(lon=-17.8905595556 * u.deg, lat=28.7613094808 * u.deg, height=2186.764 * u.m),
        rough_area=17**2 * u.m**2,
        tel_id=3,
    ),
}

s = EventSource('./test_data/mc/simtel_theta_20_az_180_gdiffuse_10evts.simtel.gz')
print(s.subarray.tel_coords)

for l in locations.values():
    l['area'] = s.subarray.tel[l['tel_id']].optics.mirror_area
    l['simtel_pos'] = s.subarray.positions[l['tel_id']].to_value(u.m)


for key in ('area', 'rough_area'):
    areas = [l[key].to_value(u.m**2) for l in locations.values()]
    ref_lon = np.average([l['loc'].lon.deg for l in locations.values()], weights=areas)
    ref_lat = np.average([l['loc'].lat.deg for l in locations.values()], weights=areas)

    print(f'Reference location ({key}): {ref_lon:.6f}°, {ref_lat:.6f}°')


    crs = CRS.from_dict(dict(proj='tmerc', ellps='WGS84', lon_0=ref_lon, lat_0=ref_lat, axis='enu', units='m'))

    proj = Transformer.from_crs(crs.geodetic_crs, crs)

    print("PROJ coordinates")
    for label, d in locations.items():
        y, x = proj.transform(d['loc'].lon.deg, d['loc'].lat.deg)
        print(f"{label:10s} {x: 7.3f} m {y: 7.3f} m")
        sx, sy, sz = d['simtel_pos']
        sy = -sy
        print(f'dx={x - sx:.2f}, {y - sy:.2f}')

    print("Astropy Coordinates")
    ref = EarthLocation(lon=ref_lon * u.deg, lat=ref_lat * u.deg, height=2199 * u.m)
    ref_cart = ref.get_itrs().cartesian

    fig, ax = plt.subplots()

    print(f'Reference location: {ref_lon:.6f}°, {ref_lat:.6f}°')

    for i, (label, d) in enumerate(locations.items()):
        cart = d['loc'].get_itrs().cartesian
        altaz = ITRS(cart - ref_cart, location=ref).transform_to(AltAz(location=ref))
        x, y, z = altaz.cartesian.xyz.to_value(u.m)

        print(f"{label:10s} {x: 7.3f} m {y: 7.3f} m")

        ax.plot(x, y, 'o', color='w', mec=f'C{i}', markeredgewidth=2, label=f'{label} (astropy)')
        sx, sy, sz = d['simtel_pos']
        sy = -sy
        print(f'dx={x - sx:.2f}, {y - sy:.2f}')
        ax.plot(sx, sy, 'o', color='k', mec=f'C{i}', markeredgewidth=2, label=f'{label} (sim-config)')


    ax.legend(ncol=3, loc='lower center', bbox_to_anchor=(0.5, 1.02))

    plt.show()
```

**Comment by maxnoe** [comment]
```
In [1]: from ctapipe.io import EventSource

In [2]: obsfile = EventSource('./test_data/real/R0/20200218/LST-1.1.Run02005.0000_first50.fits.fz')

In [3]: simfile = EventSource('./test_data/mc/simtel_theta_20_az_180_gdiffuse_10evts.simtel.gz')

In [4]: simfile.subarray.positions[1]
Out[4]: <Quantity [-6.336, 60.405, 12.5  ] m>

In [5]: obsfile.subarray.positions[1]
Out[5]: <Quantity [-5.8637701 , 60.37671721,  0.88271184] m>
```

**Review by FrancaCassol — APPROVED** [review]

**Pull request merged**